src/openassetio-core/tests/hostApi/ManagerTest.cpp

```diff
@@ -243,27 +247,38 @@ SCENARIO("Resolving entities") {
 
       WHEN("singular resolve is called with default errorPolicyTag") {
         THEN("an exception is thrown") {
-          CHECK_THROWS_MATCHES(manager->resolve(ref, traits, resolveAccess, context),
-                               openassetio::MalformedEntityReferenceBatchElementException,
-                               Catch::Message("Error Message"));
+          CHECK_THROWS_MATCHES(
+              manager->resolve(ref, traits, resolveAccess, context),
+              openassetio::errors::BatchElementException,
+              Catch::Predicate<openassetio::errors::BatchElementException>(
```

Comment:
[consider] Factor out this handy `Catch::Predicate` object (or factory for it), since it's duplicated quite a bit.

Reply:
Nice